check_run/tests/setup.py

```diff
@@ -785,3 +796,92 @@ def create_manual_payment_entry(settings):
 
 	pe.save()
 	pe.submit()
+
+
+def create_customers(settings=None):
+	for customer in customers:
+		c = frappe.new_doc("Customer")
+		c.customer_name = customer[0]
+		c.customer_group = "Commercial"
+		c.customer_type = "Company"
+		c.territory = "United States"
+		c.save()
+
+		addr = frappe.new_doc("Address")
+		addr.address_title = f"{customer[0]} - {customer[1]['city']}"
+		addr.address_type = "Billing"
+		addr.address_line1 = customer[1]["address_line1"]
+		addr.city = customer[1]["city"]
+		addr.state = customer[1]["state"]
+		addr.country = customer[1]["country"]
+		addr.pincode = customer[1]["pincode"]
+		addr.append("links", {"link_doctype": "Customer", "link_name": c.name})
+		addr.save()
+
+
+def modify_tax_templates(settings=None):
+	frappe.db.sql_ddl("truncate `tabSales Taxes and Charges Template`")
+	frappe.db.sql_ddl("truncate `tabPurchase Taxes and Charges Template`")
+	frappe.db.sql_ddl("truncate `tabItem Tax Template`")
+	# frappe.delete_doc("Account", "ST 6.25% - CFC")
+	# frappe.delete_doc("Account", "ST 4% - CFC")
+	for st in sales_tax_templates:
+		frappe.get_doc(**st).insert()
+
+
+def create_sales_invoices(settings):
+	item = frappe.new_doc("Item")
+	item.item_code = "Cloudberry"
+	item.item_group = "Products"
+	item.stock_uom = "Nos"
+	item.maintain_stock = 1
+	item.is_sales_item = 1
+	item.is_sub_contracted_item, item.include_item_in_manufacturing = 0, 0
+	item.grant_commission = 0
+	item.is_purchase_item = 1
+	item.append(
+		"item_defaults",
+		{"company": settings.company, "default_warehouse": ""},
+	)
+	item.save()
+
+	se = frappe.new_doc("Stock Entry")
+	se.posting_date = settings.day
+	se.set_posting_time = 1
+	se.stock_entry_type = "Material Receipt"
+	se.append(
+		"items",
+		{
+			"item_code": "Cloudberry",
+			"t_warehouse": "Stores - CFC",
+			"qty": 1000,
+			"uom": "Nos",
+			"stock_uom": "Nos",
+			"conversion_factor": 1,
+			"basic_rate": 0.65,
+			"expense_account": "1910 - Temporary Opening - CFC",
+		},
+	)
+	se.save()
+	se.submit()
+
+	for customer in customers:
+		si = frappe.new_doc("Sales Invoice")
+		si.customer = customer[0]
+		si.set_posting_time = 1
+		si.company = settings.company
+		si.posting_date = settings.day
+		si.append("items", {"item_code": "Cloudberry", "qty": 100, "rate": 1.30})
+		si.taxes_and_charges = "MA Sales Tax - CFC"
+		# this API is typically only called from the browser
+		taxes = frappe.call(
+			"erpnext.controllers.accounts_controller.get_taxes_and_charges",
+			**{
+				"master_doctype": "Sales Taxes and Charges Template",
+				"master_name": si.taxes_and_charges,
+			},
+		)
+		for tax in taxes:
+			si.append("taxes", tax)
+		si.save()
+		si.submit()

```

Comment:
Add at least one Credit Note to test net payable

Reply:
https://github.com/agritheory/check_run/pull/231